source/AI.cpp

```diff
@@ -4207,21 +4325,22 @@ void AI::IssueOrders(const PlayerInfo &player, const Orders &newOrders, const st
 
 	// A target is valid if we have no target, or when the target is in the
 	// same system as the flagship.
-	bool isValidTarget = !newTarget || (newTarget && player.Flagship() &&
-		newTarget->GetSystem() == player.Flagship()->GetSystem());
+	bool isValidTarget = !targetShip || targetAsteroid
```

Comment:
This variable is true if targetShip is false. Can targetAsteroid and targetShip both be true at the same time? If not, is this second variable not always short circuited by the first?

Reply:
I don't believe a ship can target both an asteroid and a ship at the same time.  It is one or the other.

If a ship is not targeted `isValidTarget` is false and this sets it to `true` if the ship has targeted an asteroid.  This is necessary for the fleet to chase because a ship will not chase if the target is invalid.